mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/TaskExecutorRegistration.java

```diff
@@ -40,4 +41,11 @@ public class TaskExecutorRegistration {
 
   // machine information identifies the cpu/mem/disk/network resources of the task executor.
   MachineDefinition machineDefinition;
+
+  // custom attributes describing the task executor
+  Map<String, String> taskExecutorAttributes;
+
+  public boolean containsAttributes(Map<String, String> requiredAttributes) {
+      return taskExecutorAttributes.entrySet().containsAll(requiredAttributes.entrySet());
```

Comment:
Null check? in #239 I didn't mark this one as nonNull yet for back-compat on existing workers.

Reply:
It should never be null. We ensure that even if the taskExecutorAttributes are not found in the JSON, the JsonSerializer should deserialize this field as an empty map.